### ISSUE TITLE
Broken codes and cannot reproduce reported results after repair

### ISSUE THREAD

**Issue by machengcheng2016** [msfuxian/MUSIC#3]
Greetings,
This paper is really cool and it's very nice of authors to release the codes. However, the codes seem incomplete, redundant, and unable to run. For example, 
1. There is lack of evaluation code; 
2. There are a few useless variables and functions, for example, [`_unlabel_t `](https://github.com/msfuxian/MUSIC/blob/main/utils.py#L129) and [`class_num`](https://github.com/msfuxian/MUSIC/blob/main/utils.py#L175) are never used, [`train_loop`](https://github.com/msfuxian/MUSIC/blob/main/utils.py#L43) and [`train_loop_NL`](https://github.com/msfuxian/MUSIC/blob/main/utils.py#L61) are totally the same;
3. There are a few typos, for example, `nl_pred4` is miswritten as [`nl_pred3`](https://github.com/msfuxian/MUSIC/blob/main/utils.py#L172);
4. There are some wrongly assigned values, for example, [`in_channels=18432`](https://github.com/msfuxian/MUSIC/blob/main/utils.py#L119) is not true for the ResNet-12 backbone, which should be 512 instead. The [`250`](https://github.com/msfuxian/MUSIC/blob/main/utils.py#L115) should be decided by `unlabel_embeddings.shape[0]`, which should be 150 in 5-way-1-shot setting instead;
5. The [`NL_loss`](https://github.com/msfuxian/MUSIC/blob/main/loss.py#L3) is different defined from Eq. (5) in original paper.

After repairing all these mistakes, I still found that the performances reported by paper cannot be reproduced. During experiments, as long as I use the `NL_loss` on unlabeled data, the test set accuracy drops to less than 60% on miniImageNet, no matter how I grid-search the learning rate or iteration number. In comparison, if I only use the labeled data to train classifier, I got 61% on miniImageNet, which is absolutely far below 74.86% as reported in paper.
Furthermore, in order to observe whether `NL_loss` is always harmful to performance, I try to use the `NL_loss` the labeled data (`loss = ce_loss + entropy_loss + weight * nl_loss`). Again, no matter how I grid-search the loss weight, learning rate or iteration, the performances is always inferior to that of (`loss = ce_loss + entropy_loss`).

without NL_loss
![image](https://github.com/msfuxian/MUSIC/assets/22851860/b528416d-155e-481f-8ebf-8da4e014395c)

with NL_loss
![image](https://github.com/msfuxian/MUSIC/assets/22851860/a265bf66-ca95-4712-a692-47d31511456b)
